File: 2026-2/sesion-de-clase-4/java/Solución/Lab01-2026-1/pe/edu/pucp/preguntas/MultipleChoiceQuestion.java
package pe.edu.pucp.preguntas;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import pe.edu.pucp.examenes.AssessmentItem;

public class MultipleChoiceQuestion extends Question {

    private List<Integer> respuestas;

    public MultipleChoiceQuestion(
            int codigo,
            String prompt,
            List<String> opciones,
            List<Integer> respuestas) {
        super(codigo, prompt, opciones);
        this.respuestas = new ArrayList<>(respuestas);
    }

    @Override
    public void devolverDatos() {
        super.devolverDatos();
        System.out.println("Seleccione las opciones de su respuesta: ");
    }

    @Override
    public void recuperarRespuesta(AssessmentItem item) {
        Scanner scanner = new Scanner(System.in);
        String respuestaAlumno = scanner.next();
        String[] resps = respuestaAlumno.split(",");
        List<Integer> respsList = new ArrayList<>();
        for (String r : resps) {
            respsList.add(Integer.parseInt(r));
        }

        item.setEsCorrecto(respuestas.equals(respsList));
    }
}

File: 2026-2/sesion-de-clase-05/dotnet/Lab01Lib/MultipleChoiceQuestion.cs
namespace PUCP.Labs.Lab01.Preguntas;

using PUCP.Labs.Lab01.Examenes;

public class MultipleChoiceQuestion : Question
{
    private readonly List<int> respuestas;

    public MultipleChoiceQuestion(
            int codigo,
            string prompt,
            List<string> opciones,
            List<int> respuestas)
        : base(codigo, prompt, opciones)
    {
        this.respuestas = [.. respuestas];
    }

    public override void DevolverDatos()
    {
        base.DevolverDatos();
        Console.WriteLine("Seleccione las opciones de su respuesta: ");
    }

    public override void RecuperarRespuesta(AssessmentItem item)
    {
        string respuestaAlumno = Console.ReadLine()!;
        string[] resps = respuestaAlumno.Split(',');
        List<int> respsList = new List<int>();
        foreach (string r in resps)
        {
            respsList.Add(int.Parse(r));
        }

        item.EsCorrecto = respuestas.SequenceEqual(respsList);
    }
}
